### PR TITLE
Initial version of a runtime-agnostic async rust API client and CLI

### PR DESCRIPTION
**Pull request by jbr**
when 0.0.15 is deployed, this can be tested with `cargo install divviup-cli --git https://github.com/divviup/divviup-api --branch client` then go to https://app.staging.divviup.org/ and get a token. Then `divviup -t <<TOKEN>> account list`

begins to address #201

### CODE REVIEW

**Comment by jbr** [comment]
Both the client and cli need significantly more documentation before considering publishing a 0.0.1 version

**Review by divergentdave — APPROVED** [review]

**Comment by jbr** [comment]
an issue with this PR that should block merging: `--buckets 1 --buckets 2 --buckets 3` is unacceptably awkward, but I'm not sure what to replace it with. `--buckets 1,2,3` and a custom parser?

File: client/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    http_client: Client,
+    headers: Headers,
+    url: Url,
+}
+
+const DEFAULT_URL: &str = "https://api.staging.divviup.org/";
+
+impl DivviupClient {
+    pub fn new(token: String, http_client: impl Into<Client>) -> Self {
+        let headers = Headers::from_iter([
```
[the remark this answers is not in the document]
Comment (jbr) [reply client/src/lib.rs:52]: added in 20c3c422e848b5b8404fc2ba5e937e6b81131a75

File: cli/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+use aggregators::AggregatorAction;
+use api_tokens::ApiTokenAction;
+use clap::{Parser, Subcommand};
+use divviup_client::{Client, DivviupClient, Url, Uuid};
+use memberships::MembershipAction;
+use tasks::TaskAction;
+use trillium_rustls::RustlsConfig;
+use trillium_tokio::ClientConfig;
+
+#[derive(Parser, Debug)]
```
[the remark this answers is not in the document]
Comment (jbr) [reply cli/src/main.rs:71]: addressed in 346d2ffbe40e8da840c75a5749bd7ffded54cfdb

File: cli/hpke
[the remark this answers is not in the document]
Comment (jbr) [reply cli/hpke:1]: addressed in 346d2ffbe40e8da840c75a5749bd7ffded54cfdb

**Pull request merged**